### PR TITLE
ITOF camera implementation

### PR DESCRIPTION
**Pull request by cristi-iacob**

### CODE REVIEW

File: sdk/src/cameras/itof-camera/mode_info.cpp [hunk 1]
```diff
@@ -9,12 +9,17 @@
 
 // TODO: load this information from camera module EEPROM
 ModeInfo::modeInfo ModeInfo::g_modeInfoData[] = {
-    {0, 512, 512, 6, 1317, 896},     {1, 320, 288, 9, 2162, 288},
-    {2, 512, 512, 3, 1317, 448},     {3, 1024, 1024, 1, 12289, 64},
-    {4, 640, 576, 10, 4321, 640},    {5, 1024, 1024, 10, 12289, 640},
-    {6, 1024, 256, 10, 2240, 878},   {7, 512, 512, 10, 2195, 896},
-    {8, 1024, 1024, 1, 12289, 64},   {9, 512, 512, 3, 1317, 448},
-    {10, 1024, 1024, 9, 12289, 576},
+    {0, 512, 512, 6, 1317, 896, 0},
```
Comment (dNechita) [inline sdk/src/cameras/itof-camera/mode_info.cpp:12]: How do you know which mode is passive IR?

Comment (cristi-iacob) [reply sdk/src/cameras/itof-camera/mode_info.cpp:12]: If I understand the code from the old SDK correctly, there are now multiple modes that will capture passive IR alongside with the depth frame. This modification defines which modes support it and which not.

Comment (dNechita) [reply sdk/src/cameras/itof-camera/mode_info.cpp:12]: The old SDK states that only mode number 3 is considered passive IR.

File: sdk/src/cameras/itof-camera/camera_itof.cpp [hunk 2]
```diff
@@ -106,22 +266,162 @@ CameraItof::getAvailableModes(std::vector<std::string> &availableModes) const {
     availableModes.emplace_back("pcm8");
     availableModes.emplace_back("ahat2");
     availableModes.emplace_back("mp");
+    availableModes.emplace_back("unknown");
```
Comment (dNechita) [inline sdk/src/cameras/itof-camera/camera_itof.cpp:269]: There shouldn't be an unknown mode.

File: sdk/src/cameras/itof-camera/camera_itof.h [hunk 3]
```diff
@@ -200,7 +201,7 @@ class CameraItof : public aditof::Camera {
     aditof::Status processFrame(uint8_t *rawFrame, uint16_t *captureData,
                                 uint8_t *head, const uint16_t embed_height,
                                 const uint16_t embed_width,
-                                aditof::FrameDetails &frameDetails);
+                                aditof::Frame *frame);
```
Comment (VLitanADI) [inline sdk/src/cameras/itof-camera/camera_itof.h:204]: Why is this necessary?

Comment (cristi-iacob) [reply sdk/src/cameras/itof-camera/camera_itof.h:204]: This change is necessary in order to be able to set the frame attribute FrameNum. 
FrameNum was a member of FrameDetails in the old SDK implementation, but in order to keep the same API for FrameDetails for all frame implementations, we decided to remove all the additional FrameDetails and store them as frame attributes.

File: sdk/src/cameras/itof-camera/mode_info.cpp (hunk 1 above)
Comment (cristi-iacob) [reply sdk/src/cameras/itof-camera/mode_info.cpp:12]: In the latest version of the old SDK there is a modification, which adds the passive IR feature for multiple modes.

File: sdk/src/cameras/itof-camera/camera_itof.cpp (hunk 2 above)
Comment (cristi-iacob) [reply sdk/src/cameras/itof-camera/camera_itof.cpp]: Solved.

**Comment by cristi-iacob** [comment]
I removed all the unnecessary defines from the aditof_internal.h file.

**Review by dNechita — APPROVED** [review]
